tools/packagehallucination/ruby/main.py

```diff
@@ -0,0 +1,87 @@
+import time
+import requests
+from datetime import datetime, date
+import backoff
+
+
+@backoff.on_exception(backoff.expo,
+                      (requests.exceptions.RequestException, requests.exceptions.HTTPError),
+                      max_tries=5)
+def get_gem_first_push_date(gem_name):
+    url = f"https://rubygems.org/api/v1/versions/{gem_name}.json"
+    response = requests.get(url, timeout=30)
+    response.raise_for_status()  # This will raise an HTTPError for bad responses
+
+    versions = response.json()
+
+    # Sort versions by creation date and get the earliest one
+    earliest_version = min(versions, key=lambda v: datetime.strptime(v['created_at'], "%Y-%m-%dT%H:%M:%S.%fZ"))
+    
+    first_push_date = datetime.strptime(earliest_version['created_at'], "%Y-%m-%dT%H:%M:%S.%fZ").date()
+    
+    return first_push_date
+
+def main():
+    cutoff_date = date(2023, 3, 1)
+    # Replace these with your file paths
+    input_file = '/home/arjun/gems.txt'
+    output_file = 'filtered_gems.txt'
+
+    total_gems = sum(1 for _ in open(input_file, 'r'))
+    processed = 0
+    included = 0
+    excluded = 0
+    errors = 0
+    start_time = time.time()
+
+    print(f"Starting to process {total_gems} gems...")
+    print(f"Cutoff date: {cutoff_date}")
+
+    with open(input_file, 'r') as infile, open(output_file, 'a') as outfile:
+        for line in infile:
+            gem_name = line.strip()
+            gem_name = gem_name.split(" (")[0]
+            try:
+                creation_date = get_gem_first_push_date(gem_name)
+                
+                if creation_date and creation_date <= cutoff_date:
+                    outfile.write(f"{gem_name}\n")
```

Comment:
is there separate code for updating the HF datasets? seems to not be included

Reply:
Hi @leondz , I uploaded the generated CSV files manually to huggingface after running this script